api/process.go

```diff
@@ -82,6 +95,98 @@ func (a *API) createProcessHandler(w http.ResponseWriter, r *http.Request) {
 	apicommon.HTTPWriteJSON(w, processID)
 }
 
+// updateProcessHandler godoc
+//
+//	@Summary		Update an existing voting process
+//	@Description	Update an existing voting process. Requires Manager/Admin role.
+//	@Tags			process
+//	@Accept			json
+//	@Produce		json
+//	@Security		BearerAuth
+//	@Param			processId	path		string							true	"Process ID"
+//	@Param			request		body		apicommon.CreateProcessRequest	true	"Process update information"
+//	@Success		200			{string}	string							"OK"
+//	@Failure		400			{object}	errors.Error					"Invalid input data"
+//	@Failure		401			{object}	errors.Error					"Unauthorized"
+//	@Failure		404			{object}	errors.Error					"Process not found"
+//	@Failure		500			{object}	errors.Error					"Internal server error"
+//	@Router			/process/{processId} [put]
+func (a *API) updateProcessHandler(w http.ResponseWriter, r *http.Request) {
+	processID := chi.URLParam(r, "processId")
+	if len(processID) == 0 {
+		errors.ErrMalformedURLParam.Withf("missing process ID").Write(w)
+		return
+	}
+
+	// parse the process info from the request body
+	processInfo := &apicommon.UpdateProcessRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&processInfo); err != nil {
+		errors.ErrMalformedBody.Write(w)
+		return
+	}
+
+	// get the user from the request context
+	user, ok := apicommon.UserFromContext(r.Context())
+	if !ok {
+		errors.ErrUnauthorized.Write(w)
+		return
+	}
+
+	existingProcess, err := a.db.Process(processID)
+	if err != nil {
+		if err == db.ErrNotFound {
+			errors.ErrMalformedURLParam.Withf("process not found").Write(w)
+			return
+		}
+		errors.ErrGenericInternalServerError.WithErr(err).Write(w)
+		return
+	}
+
+	// check the user has the necessary permissions
+	if !user.HasRoleFor(existingProcess.OrgAddress, db.ManagerRole) &&
+		!user.HasRoleFor(existingProcess.OrgAddress, db.AdminRole) {
+		errors.ErrUnauthorized.Withf("user is not admin of organization").Write(w)
+		return
+	}
+
+	var census *db.Census
+	if processInfo.CensusID != nil {
+		census, err = a.db.Census(processInfo.CensusID.String())
+		if err != nil {
+			if err == db.ErrNotFound {
+				errors.ErrMalformedURLParam.Withf("census not found").Write(w)
+				return
+			}
+			errors.ErrGenericInternalServerError.WithErr(err).Write(w)
+			return
+		}
+	}
+
+	if processInfo.Draft != nil {
+		existingProcess.Draft = *processInfo.Draft
+	}
```

Comment:
existingProcess.Draft is not used at all, i.e. not checked before allowing update, so a user can update a process with PUT even after pushing to Vochain, AFAICS?